File: dsa-ts/linear-search-int/ts/search_test.ts
import { assertEquals } from "@std/assert";
import { search } from "./search_v1.ts";

Deno.test("search()", async (t) => {
  await t.step("should find nothing if input is empty", () => {
    assertEquals(search(1, []), false);
  });

  await t.step("should find if anywhere in the array", () => {
    assertEquals(search(1, [1]), true);
    assertEquals(search(1, [-5, 1, 3]), true);
    assertEquals(search(1, [-5, 3, 1]), true);
  });
});

File: dsa-ts/linear-search-int/go/search_test.go
package linearsearchintv1_test

import (
	"testing"
	"github.com/stretchr/testify/require"
	v1 "dsago/linear-search-int/go/v1"
)

func TestSearch(t *testing.T) {
	t.Run("should find nothing if input is empty", func(t *testing.T) {
		require.False(t, v1.Search(1, []int{}))
	})

	t.Run("should find if anywhere in the array", func(t *testing.T) {
		require.True(t, v1.Search(1, []int{1}))
		require.True(t, v1.Search(1, []int{-5, 1, 3}))
		require.True(t, v1.Search(1, []int{-5, 3, 1}))
	})
}
